src/main/java/zone/cogni/asquare/security2/permission/PermissionService.java

```diff
@@ -0,0 +1,128 @@
+package zone.cogni.asquare.security2.permission;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.GrantedAuthority;
+
+import javax.annotation.PostConstruct;
+import java.io.InputStream;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+public class PermissionService<T extends Enum> {
+  private static final Logger log = LoggerFactory.getLogger(PermissionService.class);
+  private final Map<String, Set<String>> blacklistProperties = new HashMap<>();
+  private final Map<String, Set<T>> roleName2permissions = Collections.synchronizedMap(new HashMap<>());
+  private final Class<T> enumClass;
+  private Set<T> defaultPermissions;
+  @Value("${cognizone.permission.roles.resource:/role_access.json}")
+  private String roleAccess;
+
+  public PermissionService(Class<T> enumClass) {
+    this.enumClass = enumClass;
+  }
+
+  @PostConstruct
+  @SuppressWarnings("unchecked")
+  private void init() throws Exception {
+    log.info("Reading properties of {}", roleAccess);
+    try (InputStream inputStream = PermissionService.class.getResourceAsStream(roleAccess)) {
```

Comment:
don't load resource like this, it limits you to classPath resources
note, you can directly make spring inject a Resource instead of a String
then to get InputStream you can use ResourceHelper.getInputStream(...)